### PR TITLE
Simplified type creation

### PR DESCRIPTION
**Pull request by Jarodshow**
- Create `GenericOpenVRController` based on type

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs [hunk 1]
```diff
@@ -206,29 +206,11 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
...
-                case SupportedControllerType.OculusRemote:
-                    detectedController = new OculusRemoteController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-            }
+            GenericOpenVRController detectedController = Activator.CreateInstance(controllerType, 
+                new object[] { TrackingState.NotTracked, controllingHand, inputSource }) as GenericOpenVRController;
 
             Debug.Assert(detectedController != null);
-            detectedController?.SetupConfiguration(controllerType);
+            detectedController.SetupConfiguration(controllerType);
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: Let's keep the null check here.

Even if unity throws the assert, it'll still crash unfortunately :/

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs [hunk 2]
```diff
@@ -206,29 +206,11 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
...
-                    break;
-                case SupportedControllerType.OculusTouch:
-                    detectedController = new OculusTouchController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-                case SupportedControllerType.OculusRemote:
-                    detectedController = new OculusRemoteController(TrackingState.NotTracked, controllingHand, inputSource);
-                    break;
-            }
+            GenericOpenVRController detectedController = Activator.CreateInstance(controllerType, 
+                new object[] { TrackingState.NotTracked, controllingHand, inputSource }) as GenericOpenVRController;
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: Does this still allow OpenVR apps to work with the Windows Mixed Reality controllers when using the WMR driver for Steam?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: I'll pull and check now.

**Comment by StephenHodgson** [comment]
@Jarodshow just a little tip/trick. Be sure to branch your work in your repository so it's easier to pull, and easier to identify before we pull it. Thanks for adding this!

**Comment by Jarodshow** [comment]
@StephenHodgson I'll do that in the future :)

**Comment by StephenHodgson** [comment]
Getting errors:

`MissingMethodException: Constructor on type 'Microsoft.MixedReality.Toolkit.Internal.Devices.OpenVR.GenericOpenVRController' not found.`

**Comment by Jarodshow** [comment]
missingmethodexception
```
public GenericOpenVRController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
                : base(trackingState, controllerHandedness, inputSource, interactions) { }
```

I guess we should specify the default arguments?

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs (hunk 2 above)
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: You can simplify this whole line down to:
```


var detectedController = Activator.CreateInstance(controllerType, TrackingState.NotTracked, controllingHand, inputSource, null) as GenericOpenVRController;
```

**Comment by StephenHodgson** [comment]
yeah you need to add the null on the end.

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs [hunk 3]
```diff
@@ -207,7 +207,7 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
             var inputSource = InputSystem?.RequestNewGenericInputSource($"{CurrentControllerType} Controller {controllingHand}", pointers);
 
             GenericOpenVRController detectedController = Activator.CreateInstance(controllerType, 
-                new object[] { TrackingState.NotTracked, controllingHand, inputSource }) as GenericOpenVRController;
+                new object[] { TrackingState.NotTracked, controllingHand, inputSource, null }) as GenericOpenVRController;
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: The array declaration is also redundant.

**Comment by StephenHodgson** [comment]
Thanks @Jarodshow!

**Pull request merged**